### PR TITLE
opae: fpgainfo to support DFL driver patchset2

### PR DESCRIPTION
**Pull request by anandaravuri**
-Refactor sysfs path
 -Modify fpgainfo to support DFL driver

### CODE REVIEW

File: tools/base/fpgainfo/board.c [hunk 1]
```diff
@@ -400,15 +411,18 @@ fpga_result fpgainfo_board_info(fpga_token token)
 
 	res = load_board_plugin(token, &dl_handle);
 	if (res != FPGA_OK) {
-		OPAE_ERR("Failed to load board plugin\n");
+		OPAE_MSG("Failed to load board plugin\n");
```
Comment (tswhison) [inline tools/base/fpgainfo/board.c]: Seems like this would be an error?

File: tools/base/fpgainfo/board.c [hunk 2]
```diff
@@ -400,15 +411,18 @@ fpga_result fpgainfo_board_info(fpga_token token)
...
 		goto out;
 	}
 
+	if(dl_handle == NULL)
+		return FPGA_OK;
```
Comment (tswhison) [inline tools/base/fpgainfo/board.c]: Why FPGA_OK if we failed to load the library?

File: tools/base/fpgainfo/board.c [hunk 3]
```diff
@@ -400,15 +411,18 @@ fpga_result fpgainfo_board_info(fpga_token token)
...
+
 	print_board_info = dlsym(dl_handle, "print_board_info");
 	if (print_board_info) {
 		res = print_board_info(token);
 	} else {
-		OPAE_ERR("No print_board_info entry point:%s\n", dlerror());
+		OPAE_MSG("No print_board_info entry point:%s\n", dlerror());
```
Comment (tswhison) [inline tools/base/fpgainfo/board.c]: Isn't this an error?

File: tools/base/fpgainfo/board.c [hunk 4]
```diff
@@ -427,15 +441,15 @@ fpga_result mac_info(fpga_token token)
...
+		OPAE_MSG("Failed to load board plugin\n");
 		goto out;
 	}
 
 	print_mac_info = dlsym(dl_handle, "print_mac_info");
 	if (print_mac_info) {
 		res = print_mac_info(token);
 	} else {
-		OPAE_ERR("No print_mac_info entry point:%s\n", dlerror());
+		OPAE_MSG("No print_mac_info entry point:%s\n", dlerror());
```
Comment (tswhison) [inline tools/base/fpgainfo/board.c]: Seems like this should be an error.

File: tools/base/fpgainfo/board.c [hunk 5]
```diff
@@ -454,15 +468,15 @@ fpga_result phy_group_info(fpga_token token)
...
+		OPAE_MSG("Failed to load board plugin\n");
 		goto out;
 	}
 
 	print_phy_info = dlsym(dl_handle, "print_phy_info");
 	if (print_phy_info) {
 		res = print_phy_info(token);
 	} else {
-		OPAE_ERR("No print_phy_info entry point:%s\n", dlerror());
+		OPAE_MSG("No print_phy_info entry point:%s\n", dlerror());
```
Comment (tswhison) [inline tools/base/fpgainfo/board.c]: Seems like this should be an error.

File: tools/base/fpgainfo/tempinfo.c [hunk 6]
```diff
@@ -56,28 +56,42 @@ static void print_temp_info(fpga_token token)
...
 	ON_FPGAINFO_ERR_GOTO(res, out_destroy, "Failure reading properties from token");
 
 	fpgainfo_board_info(token);
 	fpgainfo_print_common("//****** TEMP ******//", props);
 
-	res = fpgaTokenGetObject(token, PKG_TEMP_NAME, &obj, FPGA_OBJECT_GLOB);
-	ON_FPGAINFO_ERR_GOTO(res, out_destroy, "Failure getting temp object from token");
-	res = fpgaObjectRead64(obj, &pkg_temp, FPGA_OBJECT_SYNC);
-	ON_FPGAINFO_ERR_GOTO(res, out_destroy, "Failure reading package temperature value");
+	if (!stat("/sys/class/fpga", &st)) {
```
Comment (tswhison) [inline tools/base/fpgainfo/tempinfo.c]: Seems like this needs an abstraction.  Why do we need to look in sysfs?

File: tools/base/fpgainfo/tempinfo.c [hunk 7]
```diff
@@ -56,28 +56,42 @@ static void print_temp_info(fpga_token token)
...
+
+		res = fpgaTokenGetObject(token, PKG_TEMP_NAME, &obj, FPGA_OBJECT_GLOB);
+		ON_FPGAINFO_ERR_GOTO(res, out_destroy, "Failure getting temp object from token");
+		res = fpgaObjectRead64(obj, &pkg_temp, FPGA_OBJECT_SYNC);
+		ON_FPGAINFO_ERR_GOTO(res, out_destroy, "Failure reading package temperature value");
+		printf("%-32s : %02ld %s\n", "Package Temperature", pkg_temp, "Centigrade");
+	}
+
+	if (!stat("/sys/class/fpga_region", &st)) {
```
Comment (tswhison) [inline tools/base/fpgainfo/tempinfo.c]: Need an abstraction here?  Why do we need to look in sysfs?

File: tools/base/libboard/board_vc/board_vc.h [hunk 8]
```diff
@@ -195,6 +195,17 @@ fpga_result parse_fw_ver(char *buf, char *fw_ver, size_t len);
 */
 fpga_result get_phy_info(char *dev_path, fpga_phy_group_info *info);
 
+
+/**
+* Prints pkvl informantion..
```
Comment (tswhison) [inline tools/base/libboard/board_vc/board_vc.h]: s/informantion/information/

File: tools/base/libboard/board_vc/board_vc.h [hunk 9]
```diff
@@ -195,6 +195,17 @@ fpga_result parse_fw_ver(char *buf, char *fw_ver, size_t len);
...
+*
+* @param[in] token            fpga_token object for device (FPGA_DEVICE type)
+* @returns FPGA_OK on success. FPGA_EXCEPTION if token is invlaid.
```
Comment (tswhison) [inline tools/base/libboard/board_vc/board_vc.h]: s/invlaid/invalid/

File: tools/base/libboard/board_rc/board_rc.c [hunk 10]
```diff
@@ -293,6 +294,20 @@ fpga_result print_board_info(fpga_token token)
 	int version                             = 0;
 	char pwr_down_cause[FPGA_STR_SIZE]      = { 0 };
 	char reset_cause[FPGA_STR_SIZE]         = { 0 };
+	struct stat st;
+	fpga_object bmc_object;
+
+
+	if (!stat("/sys/class/fpga_region", &st)) {
```
Comment (tswhison) [inline tools/base/libboard/board_rc/board_rc.c]: Why do we need to look into sysfs?

File: tools/base/libboard/board_rc/board_rc.c [hunk 11]
```diff
@@ -293,6 +294,20 @@ fpga_result print_board_info(fpga_token token)
...
+		res = fpgaTokenGetObject(token, SYSFS_DEVID_FILE, &bmc_object, FPGA_OBJECT_GLOB);
+		if (res != FPGA_OK) {
+			printf("Board Management Controller, microcontroller FW version: %s\n", "Not Supportd");
+			printf("Last Power down cause: %s\n", "Not Supportd");
```
Comment (tswhison) [inline tools/base/libboard/board_rc/board_rc.c]: s/Supportd/Supported/

File: tools/base/libboard/board_rc/board_rc.c [hunk 12]
```diff
@@ -293,6 +294,20 @@ fpga_result print_board_info(fpga_token token)
...
+			printf("Last Reset cause: %s\n", "Not Supportd");
```
Comment (tswhison) [inline tools/base/libboard/board_rc/board_rc.c]: s/Supportd/Supported/

File: tools/base/libboard/board_vc/board_vc.h (hunk 9 above)
Comment (anandaravuri) [reply tools/base/libboard/board_vc/board_vc.h]: fixed

File: tools/base/fpgainfo/board.c (hunk 3 above)
Comment (anandaravuri) [reply tools/base/fpgainfo/board.c]: fixed

File: tools/base/fpgainfo/board.c (hunk 2 above)
Comment (anandaravuri) [reply tools/base/fpgainfo/board.c]: removed
fixed

File: tools/base/libboard/board_vc/board_vc.h (hunk 8 above)
Comment (anandaravuri) [reply tools/base/libboard/board_vc/board_vc.h]: fixed

File: tools/base/libboard/board_rc/board_rc.c (hunk 11 above)
Comment (anandaravuri) [reply tools/base/libboard/board_rc/board_rc.c]: fixed

File: tools/base/libboard/board_rc/board_rc.c (hunk 12 above)
Comment (anandaravuri) [reply tools/base/libboard/board_rc/board_rc.c]: Fixed

File: tools/base/fpgainfo/tempinfo.c (hunk 6 above)
Comment (r-rojo) [reply tools/base/fpgainfo/tempinfo.c]: Can't we come up with a better way to do this?

Comment (anandaravuri) [reply tools/base/fpgainfo/tempinfo.c]: /sys/bus/pci/drivers/intel-fpga-pci Intel FPGA driver 

/sys/bus/pci/drivers/dfl-pci  DFL Upstream driver

we have to implement common library in tools, it provides API to get type of driver etc.

fixed

**Pull request merged**